karapace/schema_reader.py

```diff
@@ -309,6 +309,8 @@ def handle_messages(self) -> None:
         raw_msgs = self.consumer.poll(timeout_ms=self.timeout_ms)
         if self.ready is False:
             self.ready = self._is_ready()
+            if self.master_coordinator:
+                self.master_coordinator.ready(True)
```

Comment:
The logic fails if `self._is_ready()` returns `False` as this has explicit `True` value passed as readiness.

Comparison for `self.master_coordniator` elsewhere has `is not None` explictly.